### PR TITLE
[CI] [UI tests] [Docker example] Build and test the docker example app, streamline it for size. 

### PR DESCRIPTION
**Pull request by marcdumais-work**
This PR adds a GitHub workflow to build and test the example trace viewer Docker application/appliance

The tests were added by integrating `theia-playwright-template` (*), and doing a bit of customising for the local integration. The result is a minimal UI test suite, that can be used on both the browser and Docker example applications (Electron support is also possible but was left-out for now). This is just the beginning of a test suite, that could be greatly enhanced, but it's better than nothing, specially to validate that the Docker example does not get badly broken without our knowledge.

Along with the above, the Docker example app was streamlined, both in how it's built and to minimize the Theia extensions it uses, resulting in a nice decrease in resulting image size, hopefully without diminishing its applicability as a Trace Viewer appliance. 

(*) https://github.com/eclipse-theia/theia-playwright-template/tree/d30db090063f40d56b1b1687851385e8ea33b599

### CODE REVIEW

File: playwright-tests/tests/theia-app.test.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/********************************************************************************
+ * Copyright (C) 2022 EclipseSource and others.
```
Comment (bhufmann) [inline playwright-tests/tests/theia-app.test.ts:2]: Shouldn't be the copyright Ericsson and others? I think you wrote this file.

2024?

File: playwright-tests/tests/theia-app.test.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
```
Comment (bhufmann) [inline playwright-tests/tests/theia-app.test.ts:14]: Should the classpath exception be removed?

File: playwright-tests/tests/page-objects/trace-explorer-view.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/********************************************************************************
+ * Copyright (C) 2024 Ericsson and others.
```
Comment (bhufmann) [inline playwright-tests/tests/page-objects/trace-explorer-view.ts:2]: Shouldn't be the copyright Ericsson and others? I think you wrote this file.

2024?

File: playwright-tests/tests/page-objects/theia-trace-viewer-app.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+/********************************************************************************
+ * Copyright (C) 2022 EclipseSource and others.
```
Comment (bhufmann) [inline playwright-tests/tests/page-objects/theia-trace-viewer-app.ts:2]: Shouldn't be the copyright Ericsson and others? I think you wrote this file.

 2024?

File: playwright-tests/tests/page-objects/trace-explorer-view.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
```
Comment (bhufmann) [inline playwright-tests/tests/page-objects/trace-explorer-view.ts:14]: Should the classpath exception be removed?

File: playwright-tests/tests/page-objects/theia-trace-viewer-app.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
```
Comment (bhufmann) [inline playwright-tests/tests/page-objects/theia-trace-viewer-app.ts:14]: Should the classpath exception be removed?

File: playwright-tests/playwright.config.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
+// *****************************************************************************
+// Copyright (C) 2021-2023 logi.cals GmbH, EclipseSource and others.
```
Comment (bhufmann) [inline playwright-tests/playwright.config.ts:2]: change 2023 to 2024

File: playwright-tests/playwright.config.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
```
Comment (bhufmann) [inline playwright-tests/playwright.config.ts:14]: Do you need the classpath exception? The original file that you based it on had it but this folder has only license EPL-2.0. Also, can config files be copyrighted?

**Review by bhufmann — CHANGES_REQUESTED** [review]

**Comment by bhufmann** [comment]
@marcdumais-work I added some comments to this PR.

File: playwright-tests/tests/theia-app.test.ts (hunk 1 above)
Comment (marcdumais-work) [reply playwright-tests/tests/theia-app.test.ts]: In the latest version I renamed that file to "trace-viewer-app.test.ts". The file now has two headers - the original one from the corresponding file from `theia-playwright-template` and a second one, for our changes. The second one has copyright year 2024.

File: playwright-tests/tests/page-objects/trace-explorer-view.ts (hunk 3 above)
Comment (marcdumais-work) [reply playwright-tests/tests/page-objects/trace-explorer-view.ts:2]: Yes, this is an original file. In short, I have taken since several years, to add the "and others" part preemptively when creating a new file. It's easy to miss updating this later, when another entity contributes to the file, resulting potentially in it appearing like the original copyright holder is taking credit for contributions of another. Until such other contribution happens, the git history (log) will confirm that the original copyright holder is the sole owner, and that the "others" set is empty. This way, the copyright line needs never be updated. 

TL;DR: IMHO, it's not wrong to add "and others" since it's easily demonstrated and there are no others, but it potentially becomes wrong not to have the mention (i.e. to forget to add it), when there demonstrably is at least one "other".

File: playwright-tests/tests/page-objects/theia-trace-viewer-app.ts (hunk 4 above)
Comment (marcdumais-work) [reply playwright-tests/tests/page-objects/theia-trace-viewer-app.ts:2]: In the latest version, I have kept the original copyright header as-is (like above) and added a second header that has "Ericsson and others" as owner and the current year.

File: playwright-tests/tests/page-objects/trace-explorer-view.ts (hunk 5 above)
Comment (marcdumais-work) [reply playwright-tests/tests/page-objects/trace-explorer-view.ts]: yes, good catch!

Comment (marcdumais-work) [reply playwright-tests/tests/page-objects/trace-explorer-view.ts]: done is latest version

File: playwright-tests/tests/page-objects/theia-trace-viewer-app.ts (hunk 6 above)
Comment (marcdumais-work) [reply playwright-tests/tests/page-objects/theia-trace-viewer-app.ts:14]: Done - In the latest version, I added a second copyright header, with only EPL-2.0

File: playwright-tests/playwright.config.ts (hunk 7 above)
Comment (marcdumais-work) [reply playwright-tests/playwright.config.ts:2]: I did not modify this file, and I do not believe that we should assert copyright even if I did (since it's only a config file). I think it's safer not to modify the existing copyright header (i.e. not go against the original author's assertion of copyright).

Makes sense? Alternatively we could add a second copyright header, if we think we should assert copyright for eventual updates to this file.

Comment (bhufmann) [reply playwright-tests/playwright.config.ts:2]: Since you didn't modify the file, then it's ok to keep as-is. The second header can be added later, if needed.

File: playwright-tests/playwright.config.ts (hunk 8 above)
Comment (marcdumais-work) [reply playwright-tests/playwright.config.ts:14]: I'm probably repeating myself, but I think we should not modify the original copyright header. However, in the context of this repository, its effective license is indeed EPL-2.0 (as would be reflected if we decided to add a second copyright header for our eventual modifications in this file).

**Comment by marcdumais-work** [comment]
> @marcdumais-work I added some comments to this PR.

Bernd, I made some improvements following your comments, and tried to answer your questions.

File: playwright-tests/playwright.config.ts (hunk 8 above)
Comment (bhufmann) [reply playwright-tests/playwright.config.ts:14]: @marcdumais-work  could you please reply to this one?

File: README.md [hunk 9]
```diff
@@ -420,6 +420,9 @@ git add RELEASE && git commit --amend
 
 Finally, push the branch and use it to create a PR. When the PR is merged, a GitHub release should be created with auto-generated release notes, as well as a git tag. Then the `publish-latest` CI job should trigger and if everything goes well, publish the new version of the repo's packages to `npm`.
```
Comment (bhufmann) [inline README.md:421]: Could you please add a description under the chapter `Tests` how to run the ui tests?

**Review by bhufmann — CHANGES_REQUESTED** [review]

File: README.md (hunk 9 above)
Comment (marcdumais-work) [reply README.md:440]: done

Comment (marcdumais-work) [reply README.md:440]: oops - hold-on I just noticed something wrong. Will push an update soon

**Review by bhufmann — DISMISSED** [review]
Looks good to me. Thanks for adding these ui tests.

File: README.md (hunk 9 above)
Comment (marcdumais-work) [reply README.md:440]: Ok, should be good now. Since I was updating the tests chapter, I made some corrections/enhancements to the test coverage sub-section - I had observed that the provided command resulted in no coverage report.

**Review by bhufmann — APPROVED** [review]

**Comment by marcdumais-work** [comment]
Thanks for the review!
